src/lib/lite_base/nat.ml

```diff
@@ -15,7 +15,19 @@ module type Input_intf = sig
 end
 
 module Make (M : Input_intf) = struct
```

Comment:
`Make` is asking for a `bin_io` module, and the choice of that input gives new a particular version.

So I would write:
```
module Input_32 = struct
  module Stable = struct
    module V1 = struct
      module T = struct
        type t = ...Int32.t
      end
       include T
   end
end
```